cluster/gce/container-linux/configure-helper.sh

```diff
@@ -690,7 +734,12 @@ function start-kube-proxy {
   fi
   local container_env=""
   if [[ -n "${ENABLE_CACHE_MUTATION_DETECTOR:-}" ]]; then
-    container_env="env:\n    - name: KUBE_CACHE_MUTATION_DETECTOR\n    value: \"${ENABLE_CACHE_MUTATION_DETECTOR}\""
+    # Daemonset manifest needs a different indent.
+    if [ ${is_daemonset_manifest} = true ]; then
```

Comment:
use [[ and double quote variable expansions. See test and quoting

https://google.github.io/styleguide/shell.xml

Reply:
Logic is removed. See the new commit on #50705.